controllers/sr_controller/sr_controller.py

```diff
@@ -7,25 +7,66 @@
 # Root directory of the SR webots simulator (equivalent to the root of the git repo)
 ROOT = Path(__file__).resolve().parent.parent.parent
 
-ROBOT_FILE = ROOT.parent / "robot.py"
+MODE_FILE = ROOT / "robot_mode.txt"
 
 EXAMPLE_CONTROLLER_FILE = ROOT / "controllers/example_controller/example_controller.py"
 
+ROBOT_IDS_TO_CORNERS = {
+    "291": 0,
+    "684": 1,
+    "1077": 2,
+    "1470": 3,
+}
 
-if __name__ == "__main__":
-    if not ROBOT_FILE.exists():
-        print("Robot controller not found, copying example into place.")
-        copyfile(str(EXAMPLE_CONTROLLER_FILE), str(ROBOT_FILE))
 
-    # Ensure the python path is properly passed down so the `sr` module can be imported
+def get_robot_zone() -> int:
+    return ROBOT_IDS_TO_CORNERS[os.environ['WEBOTS_ROBOT_ID']]
+
+
+def get_robot_file() -> Path:
+    zone_id = get_robot_zone()
+    robot_file_dir = ROOT.parent / ("zone-" + str(zone_id))
+    if robot_file_dir.is_dir():
+        return robot_file_dir / "robot.py"
+    if get_robot_mode() == "comp":
+        print("WARNING: Using default robot file location in competition mode.")
```

Comment:
Can we simplify this logic?

I suggest that in `comp` mode we don't fall back to the default location -- I can see that causing weirdness during the actual competition otherwise.

Reply:
It currently prints a warning. In theory it should never happen if the simulator is run in an isolated environment. 

The logic here is a little complex, but I think it's necessary to keep usability simple. Not copying the default controller would simplify things a little, but I think we definitely want to be doing that!

Reply:
Not sure where we got to with this, especially given the discussion at https://github.com/srobo/docs/pull/126#discussion_r436400754?

Anyway, I've implemented that logic as it seemed not to meet with any objections: https://github.com/srobo/competition-simulator/pull/106

Reply:
acf9e5b6e33d6c391b70211c53fd18cdd6070e67 (merges https://github.com/srobo/competition-simulator/pull/106) reworked this substantially.